### PR TITLE
add declining balance depreciation method

### PR DESCRIPTION
**Pull request by theyeeman**

### CODE REVIEW

File: pfinance/depreciation.py [hunk 1]
```diff
@@ -104,3 +104,45 @@ def units_of_production_depreciation(
             depreciation (float): The depreciation of the asset for a single period
     '''
     return (units_produced / useful_life) * (purchase_price - salvage_value)
+
+
+def declining_balance(
+    purchase_price: float,
+    salvage_value: float,
+    useful_life: int,
+    sub_periods: int = 12
```
Comment (j-pettit) [inline pfinance/depreciation.py:113]: comma

File: pfinance/depreciation.py [hunk 2]
```diff
@@ -104,3 +104,45 @@ def units_of_production_depreciation(
...
+) -> dict[str, list[float]]:
+    '''
+    Calculates the depreciation of an asset using declining balance.
+
+        Parameters:
+            purchase_price (float): The total amount paid for the asset
+            salvage_value (float): The value of the asset after its useful life
+            useful_life (int): The expected lifespan of an asset, must be greater than 0
+            sub_periods (int): The number of sub periods inside the first period (e.g. 12 months in one year). Must be less
```
Comment (j-pettit) [inline pfinance/depreciation.py:122]: default of 1 might be more intuitive, your call though

File: pfinance/depreciation.py [hunk 3]
```diff
@@ -104,3 +104,45 @@ def units_of_production_depreciation(
...
+                               than or equal to 12, default 12
+
+        Returns:
+            declining_balance_result (dict):
+                asset_value (list[float]): Value of the asset at the beginning of the period
+                periodic_depreciation (list[float]): Depreciation of the asset at the end of the period
+    '''
+    rate = round(1 - (salvage_value / purchase_price) ** (1 / useful_life), 3)
```
Comment (j-pettit) [inline pfinance/depreciation.py:130]: why round?

File: pfinance/depreciation.py [hunk 4]
```diff
@@ -104,3 +104,45 @@ def units_of_production_depreciation(
...
+    asset_value = [float(purchase_price)]
+    periodic_depreciation = [0.0]
+    periodic_depreciation.append(purchase_price * rate * sub_periods / 12)  # First depreciation is speical case
```
Comment (j-pettit) [inline pfinance/depreciation.py:133]: magic number

File: pfinance/depreciation.py (hunk 2 above)
Comment (theyeeman) [reply pfinance/depreciation.py:122]: Excel specifies a default of 12

File: pfinance/depreciation.py (hunk 3 above)
Comment (theyeeman) [reply pfinance/depreciation.py:130]: Excel specifies that rate is rounded to 3 decimal places

**Review by j-pettit — APPROVED** [review]

File: pfinance/depreciation.py (hunk 4 above)
Comment (theyeeman) [reply pfinance/depreciation.py:133]: Excel's magic number

**Pull request merged**